src/v/kafka/requests/metadata_request.cc

```diff
@@ -360,11 +372,15 @@ get_topic_metadata(request_context& ctx, metadata_request& request) {
     std::vector<ss::future<metadata_response::topic>> new_topics;
 
     for (auto& topic : *request.topics) {
+        auto source_topic = model::get_source_topic(topic);
         if (auto md = ctx.metadata_cache().get_topic_metadata(
-              model::topic_namespace_view(cluster::kafka_namespace, topic));
+              model::topic_namespace_view(
+                cluster::kafka_namespace, source_topic));
             md) {
```

Comment:
what happens w.r.t. topic auto creation if the materialized topic source is not found?

Reply:
create_topic should be called, then it should fail because there would be a non valid character within the topic name